src/common/storage/socketio/websocket.js

```diff
@@ -174,6 +174,19 @@ define([
         };
 
         this.makeCommit = function (data, callback) {
+            var localCoreObjects = data.coreObjects,
+                i,
+                keys = Object.keys(data.coreObjects || {});
+
+            if (gmeConfig.storage.patchRootCommunicationEnabled && data.patchRoot) {
+                data.coreObjects = {};
+                for (i = 0; i < keys.length; i += 1) {
+                    data.coreObjects[keys[i]] = localCoreObjects[keys[i]];
+                }
+                data.coreObjects[data.commitObject.root] = data.patchRoot;
+            }
+            delete data.patchRoot; //we remove the field in either way
+
```

Comment:
I'm not sure about his part..
The commitQueue has a reference to `data`, which has a reference to `data.coreObjects`.
Even though `localCoreObjects` still contains the full root object, there is no reference to it...
So if the cache is cleared, the rootNode is not available anywhere.